lib/Doctrine/ORM/Id/AssignedGenerator.php

```diff
@@ -59,12 +59,8 @@ public function generate(EntityManager $em, $entity)
             }
 
             if (isset($class->associationMappings[$idField])) {
-                if ( ! $em->getUnitOfWork()->isInIdentityMap($value)) {
-                    throw ORMException::entityMissingForeignAssignedId($entity, $value);
-                }
```

Comment:
I think this is actually correct though - identities deriving from associations are still valid, even for un-managed entities.